archesproject/arches
# adds test coverage github action #10738

**NOTE:**  These changes should coincide with a new rule added to the arches repo that enables the latest-branch-version status check


### Types of changes


### Description of Change


This adds a test coverage github action that:

- checks the test coverage of the current branch
- compares the test coverage against the branch it's being merged into
- fails if the test coverage is lowered
- commits a new `coverage.json` if the test coverage is not lower

NOTE: this currently only works for python unit tests.

To test:

- cut a branch from [10739](https://github.com/archesproject/arches/pull/10739) 
- remove any test to see the failure of the GH action, and leave testing unchanged to see the success.

### Issues Solved

#10738
